src/embed.ts

```diff
@@ -292,6 +296,7 @@ async function _embed(
 
   const renderer = opts.renderer ?? 'svg';
   const logLevel = opts.logLevel ?? vega.Warn;
+  const logger = opts.logger && opts.logger.level ? opts.logger.level(logLevel) : VgLogger(logLevel);
```

Comment:
Is this from Vega? It's a bit odd to check for the log level in the logger to decide whether to use the logger. What if you already have a logger? Shouldn't the first check be opts.logger ??

Reply:
I'm not sure I understand what you mean. This line doesn't check for the level _value_ but for the existence of the level _method_ (mandatory for loggers to be able to work with Vega): "If there is a custom logger and it has a level method, set its level and use it; else, use the default Vega logger."

Without this check, the script could fail on `opts.logger.level(logLevel)`,  see also [this test](https://github.com/vega/vega-embed/pull/1475/files#diff-b8f3e6c023ee1423ea82fee51a7724e7214eee1ef4a2c6abae638be05efd6688R570).

Reply:
Thanks for digging into this. The code feels overly complicated now. `opts.logger && opts.logger.level` could have just been `opts.logger?.level`. What I didn't like is that you cannot pass a logger with a preset log level. 

I think we should take the `logger` or if it's not set, use the VgLogger with the log level. If the log level is set explicitly, we can set it on the logger instance. But I think we should not se the log level to the fallback when a logger is provided. Makes sense?

I think that's what Vega does in https://vega.github.io/vega/docs/api/view/#view but it would be great to check.